File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        found = False
        l = 0
        r = len(numbers) - 1
        
        while True:
            if numbers[l] + numbers[r] == target:
                return [l+1, r+1]
            if numbers[l] + numbers[r] > target:
                r -= 1 
            if numbers[l] + numbers[r] < target:
                l += 1

File: 0167-two-sum-ii-input-array-is-sorted/0167-two-sum-ii-input-array-is-sorted.java
class Solution {
    public int[] twoSum(int[] numbers, int target) {
        int l = 0; 
        int r = numbers.length - 1; 
        
        while (true) {
            if (numbers[l] + numbers[r] == target)
                return new int[]{l + 1, r + 1}; 
            if (numbers[l] + numbers[r] > target)
                r--; 
            if (numbers[l] + numbers[r] < target)
                l++; 
        }
    }
}